logx/options.go

```diff
@@ -0,0 +1,41 @@
+package logx
+
+import "io"
+
+// Option is the common type of functions that set options
+type Option func(*options)
```

Comment:
does it make sense to have a public type with a private type as a parameter? I understand why we don't want to make options public, but this looks a bit "smelly"